Lib/_pyio.py

```diff
@@ -651,10 +651,7 @@ def read(self, size=-1):
     def readall(self):
         """Read until EOF, using multiple read() call."""
         res = bytearray()
-        while True:
-            data = self.read(DEFAULT_BUFFER_SIZE)
-            if not data:
-                break
+        while (data := self.read(DEFAULT_BUFFER_SIZE)):
```

Comment:
Andrew means that Python does not require parentheses for `while` unlike C, C++, Java, Javascript, etc. So you can safely make an extra commit with replacements like this:

```suggestion
        while data := self.read(DEFAULT_BUFFER_SIZE):
```

Reply:
Somehow I had gotten the impression that they were required. I added another commit to remove them.

Reply:
The parentheses are still present and were in the separate IDLE patch.

Reply:
But note that if the assignment expression is NOT the whole `while` expression, as in
```
while (lineinfo := text.dlineinfo(index)) is not None:
```
the parentheses ARE needed.  As I found out, without parens, the above is parsed the same as
```
while lineinfo := (text.dlineinfo(index)) is not None):
```

Reply:
Yes, that's a sharp edge that I discovered too. So I figured it would be clearer, if a little bit less elegant, to leave the parentheses in.

Reply:
It isn't, though, because then folks who are well aware of how the walrus operator works are going to wonder why there's an extra set of parentheses -- ignorance?  something got left off?  what other subtle issues will the PR have?